certora/specs/Pausable.spec

```diff
@@ -0,0 +1,93 @@
+import "helpers.spec"
+import "methods/IPausable.spec"
+
+methods {
+    pause()
+    unpause()
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Function correctness: _pause pauses the contract                                                                    │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+rule pause(env e) {
+    require nonpayable(e);
+
+    bool pausedBefore = paused();
+
+    pause@withrevert(e);
+    bool success = !lastReverted;
+
+    bool pausedAfter = paused();
+    
+    // liveness
+    assert success => !pausedBefore, "unpause call must succeed if the contract was unpaused before";
+
+    // effect
+    assert success => (pausedBefore != pausedAfter && pausedAfter == true), "contract must be paused after a successful call";

```

Comment:
we already know `!pausedBefore` from line 25, so this only needs to check 

``assert success => pausedAfter;```

Reply:
Thanks!

Solving since I took the suggestion exactly as proposed.